wp-lazy-loading.php

```diff
@@ -141,19 +138,21 @@ function wp_filter_content_tags( $content, $context = null ) {
 		$context = current_filter();
 	}
 
-	$add_srcset_sizes = 'the_content' === $context;
 	$add_loading_attr = wp_lazy_loading_enabled( 'img', $context );
 
-	if ( false === strpos( $content, '<img' ) || ( ! $add_srcset_sizes && ! $add_loading_attr ) ) {
+	if ( false === strpos( $content, '<img' ) ) {
```

Comment:
I don't have a strong opinion here, but by removing the `$add_srcset_sizes` condition we actually change the behavior of where `srcset` and `sizes` are added (currently only supported for `the_content`). This is kind of an unintended side effect (this is only about `loading`), therefore I would rather have us keep the current behavior to only do that in `the_content`. Changing that is for another discussion IMO.

If we keep it, then having this condition can avoid unnecessary execution of the heavier logic below.

Reply:
Yeah, went back and forth on this a few times. If we keep it, it makes `wp_filter_content_tags()` a bit harder to use for a "random" HTML string/content. It is possible to use `wp_add_lazy_load_attributes()` like that.

The other reason is that by default this will run on `the_content`, `the_excerpt`, and `widget_text_content` (after removing `comment_text`). Thinking it should have been running on both `the_excerpt`, and `widget_text_content` from when it was added, as it makes sense to add srcset, sizes, and loading attributes to images there.

Reply:
I'm okay with this change (also supporting the other filters for `srcset` and `sizes`), but we need to clarify that when talking about the core patch (and if we keep it also in the dev note).
From a technical standpoint it makes sense, I just wonder whether it will result in unrelated discussion and overcomplicate these efforts. But we can change still change it back if undesirable per core patch discussion.

Reply:
> we need to clarify that when talking about the core patch
> ...
> I just wonder whether it will result in unrelated discussion

Yeah, might happen. Thinking to separate this into another trac ticket. Just want to avoid hard-codding it there.